### DISCUSSION TITLE
blog/langchain-pinecone-chat-with-my-blog

### DISCUSSION THREAD

**Discussion by giscus** [zackproser/portfolio#610] [Announcements]
# blog/langchain-pinecone-chat-with-my-blog

You can chat with my writing and ask me questions I've already answered even when I'm not around

https://zackproser.com/blog/langchain-pinecone-chat-with-my-blog

**Comment by ArturHorbenko** [comment]
Cool stuff, thanks for sharing. 
Do you consider Pinecode the best way to go for a start, is there any self-hosted solutions for this ?

**Reply by zackproser** [reply]
Thanks so much for your comment! 

There are a ton of good vector database options available today - from FAISS (Facebook AI Similarity Search), Weaviate, PG vector, Timescale, PGAI, Turbopuffer...Redis...the list goes on and is expanding regularly. 

A bunch of these you can host yourself ^ - the benefit of Pinecone is that it's all done via API - so it's often easier to get started, and you don't need to worry about data migration, etc once you've deployed. 

Hope that helps!

**Reply by zackproser** [reply]
By the way, you can check out: https://zackproser.com/vectordatabases for inspiration / more leads.

**Reply by premraj-ms-hashroot** [reply]
What about chromadb for production?

**Reply by zackproser** [reply]
Chroma also works well! I chose Pinecone for this tutorial because the API makes it easy to get started and it's a solid choice - but there are indeed many other vector databases that work well, too. 

See also https://zackproser.com/vectordatabases
